pkg/controller/jobs/job/job_controller_test.go

```diff
@@ -2397,14 +2397,12 @@ func TestReconciler(t *testing.T) {
 				Key:      "tolerationkey2",
 				Operator: corev1.TolerationOpExists,
 				Effect:   corev1.TaintEffectNoSchedule,
-			}).
-				Suspend(false).
-				Obj(),
+			}).Suspend(false).Obj(),
 			wantJob: *baseJobWrapper.Clone().Toleration(corev1.Toleration{
-				Key:      "tolerationkey1",
+				Key:      "tolerationkey2",
```

Comment:
Please make sure we have an analogous test (either by adding or finding an existing) which demonstrate that node selectors are not part of the equality check.